NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -22,6 +22,8 @@ double getNodalValue(GlobalVector const& x, MeshLib::Mesh const& mesh,
 
     auto const index = dof_table.getLocalIndex(
         l, global_component_id, x.getRangeBegin(), x.getRangeEnd());
+    if (index == NumLib::MeshComponentMap::nop)
+        return 0.0;
```

Comment:
no, NaN should not be returned because this function is called from norm calculations in the same file. A problem I have is that the norm functions have a for loop for all mesh nodes and they call this function for nodes which are actually included in a DoF table, e.g. quadratic order nodes for pressure in HM. Because the norm calculation is sum up or taking the maximum, returning zero can still lead to correct norm.